### PR TITLE
#167484596 implemented CRUD of an article's comment

### PR DESCRIPTION
**Pull request by placideirandora**
#### What does this PR do?
It implements the feature of being able to comment on an article, retrieve the article's comments, delete the article's comments, and update the article's comments.

#### Description of Task to be done?
1. Adding a route for commenting on an article.
2. Adding a route for retrieving an article's comment.
3. Adding a route for updating an article's comment.
4. Adding a route for deleting an article's comment.
5. Adding a controller for commenting on an article.
6. Adding a controller for retrieving an article's comment.
7. Adding a controller for updating an article's comment.
8. Adding a controller for deleting an article's comment.
9. Adding tests for commenting on an article.
10. Adding tests for retrieving an article's comment.
11. Adding tests for updating an article's comment.
12. Adding tests for deleting an article's comment.
13. Documenting the feature with Swagger.

#### Screenshots
Screen Shot 2019-08-26 at 16 57 20
Screen Shot 2019-08-27 at 16 41 52
Screen Shot 2019-08-27 at 16 49 39
Screen Shot 2019-08-27 at 16 51 09
Screen Shot 2019-08-26 at 16 58 08
Screen Shot 2019-08-27 at 16 24 51
Screen Shot 2019-08-27 at 16 26 12
Screen Shot 2019-08-27 at 16 27 03

#### How should this be manually tested?
1. Clone this branch.
2. Do **npm install**.
3. Do **npm start**
4. Create an article.
5. Comment it with the following endpoint: **/api/v1/articles/{articleSlug}/comments**
6. Retrieve article's comments with the following endpoint: **/api/v1/articles/{articleSlug}/comments**
7. Update your comment with the following endpoint: **/api/v1/articles/{articleSlug}/comments/{commentId}**
8. Delete your comment with the following endpoint: **/api/v1/articles/{articleSlug}/comments/{commentId}**

### CODE REVIEW

File: src/controllers/articleController.js [hunk 1]
```diff
@@ -447,6 +449,53 @@ class ArticleController {
...
+      message: 'You have commented on the article',
+      articleComment: {
+        id: articleComment.id,
+        articleSlug: articleComment.articleSlug,
+        comment: articleComment.comment,
+        updatedAt: articleComment.updatedAt,
+        createdAt: articleComment.createdAt,
+        commenter: {
+          username: userName,
+          bio,
```
Comment (mkiterian) [inline src/controllers/articleController.js]: What's the use of bio here?

File: src/controllers/articleController.js [hunk 2]
```diff
@@ -447,6 +449,53 @@ class ArticleController {
...
+      articleId: id,
+      articleSlug: slugId,
+      userId: commenter,
+      comment: comment.trim(),
+    });
+
+    return res.status(201).json({
+      message: 'You have commented on the article',
+      articleComment: {
+        id: articleComment.id,
```
Comment (mkiterian) [inline src/controllers/articleController.js:486]: looks like a opportunity to use the spread operator here

**Review by mkiterian — CHANGES_REQUESTED** [review]

File: src/controllers/articleController.js (hunk 1 above)
Comment (placideirandora) [reply src/controllers/articleController.js]: I have referred to the README.
Screen Shot 2019-08-26 at 19 25 15

File: src/controllers/articleController.js (hunk 2 above)
Comment (placideirandora) [reply src/controllers/articleController.js:486]: How or why? I have never used it before!

File: src/controllers/articleController.js (hunk 1 above)
Comment (malaba6) [reply src/controllers/articleController.js]: But in the response, you did not include `following`

Comment (placideirandora) [reply src/controllers/articleController.js]: I think that one is not necessary!

**Review by mkiterian — APPROVED** [review]

**Review by mkiterian — CHANGES_REQUESTED** [review]
When I retrieve an article, I do not see its comments

**Comment by placideirandora** [comment]
> When I retrieve an article, I do not see its comments

I'm gonna implement the feedback.

**Review by harerakalex — APPROVED** [review]
Look good

**Pull request merged**